altair/vegalite/v5/api.py

```diff
@@ -345,11 +384,596 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
+        - Otherwise, requires an explicit guard to inform the type checker.
+    """
+    return obj is Undefined
+
+
+def _get_predicate_expr(p: Parameter) -> Union[str, core.SchemaBase, UndefinedType]:
+    # https://vega.github.io/vega-lite/docs/predicate.html
+    return getattr(p.param, "expr", Undefined)
+
+
+def _predicate_to_condition(
+    predicate: _PredicateType, *, empty: Union[bool, UndefinedType] = Undefined
+) -> _ConditionType:
+    condition: _ConditionType
+    if isinstance(predicate, Parameter):
+        predicate_expr = _get_predicate_expr(predicate)
+        if predicate.param_type == "selection" or _is_undefined(predicate_expr):
+            condition = {"param": predicate.name}
+            if isinstance(empty, bool):
+                condition["empty"] = empty
+            elif isinstance(predicate.empty, bool):
+                condition["empty"] = predicate.empty
+        else:
+            condition = {"test": predicate_expr}
+    elif _is_test_predicate(predicate):
+        condition = {"test": predicate}
+    elif isinstance(predicate, dict):
+        condition = predicate
+    elif isinstance(predicate, _expr_core.OperatorMixin):
+        condition = {"test": predicate._to_expr()}
+    else:
+        msg = f"condition predicate of type {type(predicate).__name__!r}"
+        raise NotImplementedError(msg)
+    return condition
+
+
+def _condition_to_selection(
+    condition: _ConditionType,
+    if_true: _StatementType,
+    if_false: _StatementType,
+    **kwargs: Any,
+) -> _SelectionType:
+    selection: _SelectionType
+    if isinstance(if_true, core.SchemaBase):
+        if_true = if_true.to_dict()
+    elif isinstance(if_true, str):
+        if isinstance(if_false, str):
+            msg = (
+                "A field cannot be used for both the `if_true` and `if_false` "
+                "values of a condition. "
+                "One of them has to specify a `value` or `datum` definition."
+            )
+            raise ValueError(msg)
+        else:
+            if_true = utils.parse_shorthand(if_true)
+            if_true.update(kwargs)
+    condition.update(if_true)
+    if isinstance(if_false, core.SchemaBase):
+        # For the selection, the channel definitions all allow selections
+        # already. So use this SchemaBase wrapper if possible.
+        selection = if_false.copy()
+        selection.condition = condition
+    elif isinstance(if_false, (str, dict)):
+        if isinstance(if_false, str):
+            if_false = utils.parse_shorthand(if_false)
+            if_false.update(kwargs)
+        selection = dict(condition=condition, **if_false)
+    else:
+        raise TypeError(if_false)
+    return selection
+
+
+class _Conditions(_TypedDict, total=False):
+    condition: _Required[List[_ConditionType]]
+    value: Any
+
+
+class _LiteralConfig(typing.NamedTuple):
```

Comment:
Wrapped as a `NamedTuple` to avoid conflicts with `then`, `otherwise` accepting `**kwargs`.
Also keeps the parameters easy to move around without [FBT001](https://docs.astral.sh/ruff/rules/boolean-type-hint-positional-argument/) issues.

Reply:
Removed in 801cccc80e7764df3d46cd0db0343f095ee78d01